main.ts

```diff
@@ -1,301 +1,322 @@
-                  await this.vnApi.downloadFile(this.fs, data.url, attachmentPath);
-                  return `- ![[${filename}]]`;
+                    const filename = getFilenameFromUrl(data.url);
+                    const attachmentPath = normalizePath(`${attachmentsPath}/${filename}`);
+                    await this.vnApi.downloadFile(this.fs, data.url, attachmentPath);
+                    return `- ![[${filename}]]`;
                 }
-              })
-            )
-          ).join('\n');
-          note += '\n';
+            }))).join('\n');
         }
 
-        if (recording.related_notes.length > 0) {
-          note += '\n## Related Notes\n';
-          note += recording.related_notes
-            .map((relatedNote) => `- [[${this.sanitizedTitle(relatedNote.title, relatedNote.created_at)}]]`)
-            .join('\n');
+        // Prepare context for Jinja template
+        const formattedPoints = points ? points.content.data.map((data: string) => `- ${data}`).join('\n') : null;
+        const formattedTodos = todo ? todo.content.data.map((data: string) => `- [ ] ${data}${this.settings.todoTag ? ' #' + this.settings.todoTag : ''}`).join('\n') : null;
+        const formattedTags = recording.tags && recording.tags.length > 0 ? recording.tags.map((tag: { name: string }) => `#${tag.name}`).join(' ') : null;
+
+        const context = {
+            title: title,
+            date: moment(recording.created_at).format(this.settings.dateFormat),
+            transcript: transcript,
+            audio_link: audioLink,
+            summary: summary ? summary.content.data : null,
+            tidy: tidyTranscript ? tidyTranscript.content.data : null,
+            points: formattedPoints,
+            todo: formattedTodos,
+            tweet: tweet ? tweet.content.data : null,
+            blog: blog ? blog.content.data : null,
+            email: email ? email.content.data : null,
+            custom: custom ? custom.content.data : null,
+            tags: formattedTags,
+            related_notes: recording.related_notes && recording.related_notes.length > 0
+                ? recording.related_notes.map((relatedNote: { title: string; created_at: string }) =>
+                    `- [[${this.sanitizedTitle(relatedNote.title, relatedNote.created_at)}]]`
+                ).join('\n')
+                : null,
+            subnotes: recording.subnotes && recording.subnotes.length > 0
+                ? recording.subnotes.map((subnote: { title: string; created_at: string }) =>
+                    `- [[${this.sanitizedTitle(subnote.title, subnote.created_at)}]]`
+                ).join('\n')
+                : null,
+            attachments: attachments,
+        };
+
+        // Render the template using Jinja
+        let note = jinja.render(this.settings.noteTemplate, context);
+
+        // Add metadata
+        const metadata = `---
+    recording_id: ${recording.recording_id}
+    duration: ${Math.floor(recording.duration / 60000) > 0
```

Comment:
let's not do inline calculations like this, prefer externalizing into a function